packages/client/src/create-deployment.ts

```diff
@@ -82,7 +86,34 @@ export default function buildCreateDeployment() {
       };
     }
 
-    const files = await hashes(fileList);
+    // Populate Files -> FileFsRef mapping
+    const workPath = typeof path === 'string' ? path : path[0];
+
+    let files;
+
+    if (clientOptions.archive === ArchiveFormat.Tgz) {
+      debug('Packing tarball');
+      fileList = fileList.map(file => file.replace(workPath, ''));
+      const tarStream = tar
+        .pack(workPath, {
+          entries: fileList,
+        })
+        .pipe(createGzip());
+      const tarBuffer: Buffer = await streamToBuffer(tarStream);
```

Comment:
```suggestion
      const tarBuffer = await streamToBuffer(tarStream);
```

The type here is redundant